meta-mender-qemu/scripts/docker/ext4_manipulator.py

```diff
@@ -35,6 +35,21 @@ def put(local_path, remote_path, rootfs, remote_path_mkdir_p=False):
     ret = proc.wait()
     assert ret == 0
 
+def symlink(target, remote_path, rootfs, remote_path_mkdir_p=False):
+    proc = subprocess.Popen(["debugfs", "-w", rootfs], stdin=subprocess.PIPE,
+                            stderr=subprocess.STDOUT)
+    if remote_path_mkdir_p:
+        # Create parent directories sequencially, to simulate a "mkdir -p" on the final dir
+        parent_dirs = list(PurePath(remote_path).parents)[::-1][1:]
+        for parent in parent_dirs:
+            proc.stdin.write(("mkdir %s\n" % parent).encode())
+    proc.stdin.write(("cd %s\n" % os.path.dirname(remote_path)).encode())
+    proc.stdin.write(("rm %s\n" % os.path.basename(remote_path)).encode())
+    proc.stdin.write(("symlink %s %s\n" % (os.path.basename(remote_path), target)).encode())
```

Comment:
Except for this line, everything else is the same as `put`... consider refactoring :wink:

Reply:
Fair point!